File: Tutor Demos/Week 6/Java/Strategy.java
enum Focus {
    Attack,
    Defence
}

public class Strategy {
    private Focus focus;
    public Strategy() {
        this.focus = Focus.Attack;
    }

    public void changeStrategy() {
        focus = (focus == Focus.Attack) ? Focus.Defence : Focus.Attack;
        System.out.println("Strategy changed to " + focus);
    }

    @Override
    public String toString() {
        return this.focus.toString();
    }
}

File: Tutor Demos/Week 6/Python/strategy.py
class Strategy:
    def __init__(self):
        self.__focus = "Attack"
    
    def change_strategy(self):
        self.__focus = "Attack" if self.__focus == "Defence" else "Attack"
        print(f"Strategy changed to {self.__focus}")
    
    def __str__(self):
        return self.__focus